### PR TITLE
[CDAP-6051] - Fixes upload plugin feature in Hydrator++

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Fixes incorrect assumption about 'properties' being required property while uploading plugin json
- Adds `plugins` to `Artifact-Plugins` header while uploading plugin JAR.

### CODE REVIEW

File: cdap-ui/app/features/hydratorplusplus/controllers/create/popovers/load-artifact-ctrl.js [hunk 1]
```diff
@@ -143,7 +151,7 @@ angular.module(`${PKG.name}.feature.hydratorplusplus`)
               });
           },
           (err) => {
-            this.errorMessage = `${err}`;
+            this.errorMessage = (err && err.data? `${err.data}` : `${err}` );
```
Comment (elfenheart) [inline cdap-ui/app/features/hydratorplusplus/controllers/create/popovers/load-artifact-ctrl.js]: nit: space before `?`

**Comment by elfenheart** [comment]
minor comment, otherwise 👍 LGTM

**Comment by elfenheart** [comment]
👍  LGTM

**Comment by ajainarayanan** [comment]
~~Working with @gsps1 to resolve the bug regarding database/mysql plugin which is used as an example to test this PR. Uploading a regular plugin (with .jar & .json) works. Its the special usecase like mysql that needs to be tested.~~

The plugin json was having issues. There were no issues with the functionality/implementation per se.

**Pull request merged**